src/Rules/DoctrineKeyValueStyleRule.php

```diff
@@ -194,7 +195,7 @@ public function processNode(Node $callLike, Scope $scope): array
                                 }
                             }
 
-                            if (! $argColumnType->isSuperTypeOf($valueType)->yes()) {
+                            if (! $argColumnType->accepts($valueType, true)->yes() || $valueType instanceof MixedType) {
```

Comment:
`accepts´ does more than you actually want.

we should stay with `isSuperTypeOf`, but we could e.g. add things like
```
if ($argColumnType->isFloat()->yes()) {
  $numberType = new UnionType([new IntegerType(), new FloatType()]);
  return !$numberType->isSuperTypeOf($valueType)->yes();
}
```

to super float|int for float.

Reply:
What is `accepts` for if not this kind of logic? There's no documentation that I can find for the method, so if you can elaborate on its intended use, that would help me understand. Thanks!

Reply:
accepts is meant to check whether a type at hand would accept another one, based on a lot of different phpstan configs (e.g. analysis level, feature flags, ..).

what we need is a way to determine whether the doctrine dbal api accepts a certain type and whether we can successfully store it in the db

Reply:
Okay, thanks for the clarification. It sounds like the best course of action for now is to go with your suggested change. I'll work on that later today. Thanks again!